llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -997,8 +997,17 @@ AArch64LoadStoreOpt::mergePairedInsns(MachineBasicBlock::iterator I,
       //   STRWui %w0, ...
       //   USE %w1
       //   STRWui kill %w1  ; need to clear kill flag when moving STRWui upwards
-      RegOp0.setIsKill(false);
-      RegOp1.setIsKill(false);
+      for (auto It = std::next(I);
```

Comment:
The comment above this code block needs some updates after this change, no?

Reply:
```suggestion
      // Since w1 is used between the stores, the kill flag on w0 is kept,
      // the one on w1 is dropped after merging:
      //   STPWi kill %w0, %w1, ...
      //   USE %w1
      for (auto It = std::next(I);
```

Do you think adding the sample merging result will be enough?